package/src/components/MessageList/hooks/useMessageList.ts

```diff
@@ -60,62 +71,90 @@ export const useMessageList = (params: UseMessageListParams) => {
   const { threadMessages } = useThreadContext();
   const messageList = threadList ? threadMessages : messages;
 
+  const filteredMessageList = useMemo(() => {
+    const filteredMessages = [];
+    for (const message of messageList) {
+      if (
+        shouldIncludeMessageInList(message, {
+          deletedMessagesVisibilityType,
+          userId: client.userID,
+        })
+      ) {
+        filteredMessages.push(message);
+      }
+    }
+    return filteredMessages;
+  }, [messageList, deletedMessagesVisibilityType, client.userID]);
+
+  const messageListPreviousAndNextMessageStore = useMemo(
```

Comment:
*nitpick*

it's a bit better to write this as:

```
const [messageListPreviousAndNextMessageStore] = useState(
() => new MessagePreviousAndNextMessageStore()
)
```

for such imperative handles.

You'll skip a memo and the store will be truly persisted (as well as stable as the constructor will run only once).